### PR TITLE
Add logFilter option and Log Filter section to the SDK doc.

### PR DESCRIPTION
**Pull request by novalisdenahi**
### Description

Add logFilter option and Log Filter section to the Java SDK doc.

### Notes for QA

Add logFilter option to the list and Log Filter section to the Java SDK doc.
Add only to the new V2 doc.

### Requirement checklist

- [x] I have validated my changes on a test/local environment.
- [ ] I have added my changes to the V1 and V2 documentations.
- [ ] I have checked the SNYK/Dependabot reports and applied the suggested changes.
- [ ] (Optional) I have updated outdated packages.

### CODE REVIEW

File: website/docs/sdk-reference/java.mdx [hunk 1]
```diff
@@ -871,6 +872,19 @@ You have the flexibility to use any slf4j implementation for logging with Config
...
 Examples fo <a href="https://github.com/configcat/java-sdk/blob/master/samples/console/src/main/resources/simplelogger.properties" target="_blank">slf4j-simple</a> and <a href="https://github.com/configcat/java-sdk/blob/master/samples/web/src/main/resources/logback.xml" target="_blank">logback</a> are available under the [Sample Apps](#sample-apps) section.
 
+### Log Filter
+
+You can set a custom log filter option by implementing the `LogFilterFunction` interface. The `apply` method provide the options to filter by `logLevel`, `eventId`, `message` or `exception`.
+If the `apply` method return with `true`, the event will be logged.
+
+```java
+    //Filter out the logs with the eventId 1001.
+    LogFilterFunction filterLogFunction = ( LogLevel logLevel, int eventId, String message, Throwable exception) -> eventId != 1001;
```
Comment (adams85) [inline website/docs/sdk-reference/java.mdx:882]:
```suggestion
    // Filter out events with id 1001 from the log.
    LogFilterFunction filterLogFunction = (LogLevel logLevel, int eventId, String message, Throwable exception) -> eventId != 1001;
```

File: website/docs/sdk-reference/java.mdx [hunk 2]
```diff
@@ -871,6 +872,19 @@ You have the flexibility to use any slf4j implementation for logging with Config
...
+
+    ConfigCatClient client = ConfigCatClient.get("#YOUR-SDK-KEY#", options -> options.logFilter(filterLogFunction));
+```
+> A complex or a badly implemented log filter can affect the SDK efficiency.
```
Comment (adams85) [inline website/docs/sdk-reference/java.mdx:886]:
```suggestion
> Please make sure that you log filter logic doesn't perform heavy computation and doesn't block the executing thread. A complex or incorrectly implemented log filter can negatively affect the performance of the SDK.
```

File: website/docs/sdk-reference/java.mdx [hunk 3]
```diff
@@ -871,6 +872,19 @@ You have the flexibility to use any slf4j implementation for logging with Config
...
+You can define a custom log filter option by implementing the `LogFilterFunction` interface. The `apply` method will be called by the _ConfigCat SDK_  each time a log event occurs (and the event passes the minimum log level specified by the `logLevel` option). That is, the `apply` method allows you to filter log events by `logLevel`, `eventId`, `message` or `exception`.
+If the `apply` method returns with `true`, the event will be logged, otherwise it will be skipped.
+
+```java
+    // Filter out events with id 1001 from the log.
+    LogFilterFunction filterLogFunction = ( LogLevel logLevel, int eventId, String message, Throwable exception) -> eventId != 1001;
+
+    ConfigCatClient client = ConfigCatClient.get("#YOUR-SDK-KEY#", options -> options.logFilter(filterLogFunction));
+```
+> Please make sure that your log filter logic doesn't perform heavy computation and dosen't block the executing thread. A complex or incorrectly implemented log filter can negatively affect the performance of the SDK.
```
Comment (adams85) [inline website/docs/sdk-reference/java.mdx:886]: typo + minor improvement to wording

```suggestion
> Please make sure that your log filter logic doesn't perform heavy computation and doesn't block the executing thread. A complex or incorrectly implemented log filter can degrade the performance of the SDK.
```

**Review by adams85 — CHANGES_REQUESTED** [review]

**Review by adams85 — APPROVED** [review]

**Comment by adams85** [comment]
Please wait with merging until @sigewuzhere takes a look at the PR too.

**Review by sigewuzhere — APPROVED** [review]

**Pull request merged**